Add Tutorial: OpenVPN setup for personal lab LAN access

jtnydv opened the pull request:
I have read and understood the Contributor's Certificate of Origin available at the end of 
https://raw.githubusercontent.com/hetzneronline/community-content/master/tutorial-template.md
and I hereby certify that I meet the contribution criteria described in it.
Signed-off-by: Jatin Yadav <jtnydv@protonmail.com>

rook59 commented on tutorials/setup-openvpn-lan-access/01.en.md:
```diff
@@ -0,0 +1,123 @@
+tags: ["openvpn", "personal-lab"]
+author: "Jatin Yadav"
+author_link: "https://github.com/jtnydv"
+author_img: "https://avatars3.githubusercontent.com/u/14368729?s=460&u=f4f0a30cb3f84681b5528f5a6a2669f456ee0b3d&v=4"
```
```suggestion
author_img: "https://avatars3.githubusercontent.com/u/14368729"

```

jtnydv replied:
Did the suggested change

rook59 commented on tutorials/setup-openvpn-lan-access/01.en.md:
```diff
@@ -0,0 +1,123 @@
+header_img: "header.jpg"
+---
+
+## OpenVPN setup for personal lab LAN access
```
```suggestion
## Introduction
```

The title is already mentioned in the title ;)

jtnydv replied:
Made the suggested changes.

rook59 commented on tutorials/setup-openvpn-lan-access/01.en.md:
```diff
@@ -0,0 +1,123 @@
+slug: "setup-openvpn-lan-access"
+date: "2021-01-19"
+title: "OpenVPN setup for personal lab LAN access"
+short_description: "This tutorial will guide you through the process of setting up an OpenVPN server so that you are able to access LAN from your local machine."
```
```suggestion
short_description: "This tutorial will guide you through the process of setting up an OpenVPN Access Server so that you are able to access LAN from your local machine."

```

jtnydv replied:
Did the suggested change

rook59 commented on tutorials/setup-openvpn-lan-access/01.en.md:
```diff
@@ -0,0 +1,123 @@
+path: "/tutorials/setup-openvpn-lan-access"
+slug: "setup-openvpn-lan-access"
+date: "2021-01-19"
+title: "OpenVPN setup for personal lab LAN access"
```
```suggestion
title: "OpenVPN Access Server setup for personal lab LAN access"

```

jtnydv replied:
Did the suggested change.

rook59 commented on tutorials/setup-openvpn-lan-access/01.en.md:
```diff
@@ -0,0 +1,123 @@
+
+Now let's set things up to ensure that we have access to the LAN, but complete traffic from our machine doesn't go through the server.
+
+First, let's go to the Network Settings under Configuration and change the "Hostname or IP Address." The value in that textbox should be the public IP address of the virtualization server.
```
> The value in that textbox should be the public IP address of the virtualization server.

What if the users IP address at home is a dynamic IP address? I guess setting up DynDNS and using a domain name instead of an IP address here should work. What do you think? Can you mention such a "workaround" for users with a dynamic IP?

jtnydv replied:
The IP Address I am talking about here is the Hetzner Server Public IP address not the user's IP address. I hope that makes it clear.

rook59 replied:
Got it. I thought you were referring to a home LAN, but you are absolutely right if this is happening on a dedicated server with virtualization.

rook59 commented on tutorials/setup-openvpn-lan-access/01.en.md:
````diff
@@ -0,0 +1,123 @@
+
+### Step 6
+
+Forward the LAN OpenVPN port to the Virtualization Server's Internet port. This can be done using `socat` and `tmux`
+
+Launch a new tmux session with tmux command; if not present, install tmux for your distro. Once in the tmux window, run the following command for port forwarding.
+
+```bash
+$ sudo socat TCP-LISTEN:$$INTERNET_PORT_NUMBER$$,fork,reuseaddr TCP:$$OPENVPN_SERVER_IP$$:$$SERVER_PORT$$
+```
````
This will be gone if the server is rebooted. What about a persistent iptables or ufw rule?

jtnydv replied:
I haven't looked into iptables. I am not that fluent with iptables, I will try to find a workaround.

rook59 replied:
Okay, would be definetely nice to have

rook59 commented on tutorials/setup-openvpn-lan-access/01.en.md:
```diff
@@ -0,0 +1,123 @@
+---
+SPDX-License-Identifier: MIT
+path: "/tutorials/setup-openvpn-lan-access"
+slug: "setup-openvpn-lan-access"
```
I think you should also mention OpenVPN Access Server in the path and slug

jtnydv replied:
Done the changes to the slug and path.

rook59 reviewed (changes requested):
Thank you for your tutorial! You are actually installing the proprietary "OpenVPN Access Server" product and not the OSS OpenVPN server software. The tutorial should clearly reflect this.
Can you please add small headlines about what you do in each step?
Please also take a look at our [tutorial template](https://github.com/hetzneronline/community-content/blob/master/tutorial-template.md) about the formatting of each step. You should replace `### Step X` with `## Step X`.
Your `NOTE` at the end should be a `Conclusion` instead.

rook59 commented on tutorials/setup-openvpn-lan-access/01.en.md:
```diff
@@ -0,0 +1,123 @@
+author_description: "A cyber security enthusiast!"
+language: "en"
+available_languages: ["en"]
+header_img: "header.jpg"
```
@leojakeson do we accept custom headers? I am not sure if this [header image](https://github.com/hetzneronline/community-content/blob/962e1cbb45e3c0a01b8eb4e770a3ccf61689d167/tutorials/setup-openvpn-lan-access/header.jpg) has some copyright

jtnydv replied:
Please have a look and let me know, if this would be acceptable. If not, I'll remove it.

leojakeson replied:
We don't yet support this feature (custom header images), though we are looking at implementing that in the future.

So for now, please remove it, thank you.

rook59 replied:
@jtnydv the custom header_img  and header.jpg still needs to be removed :)

jtnydv reviewed (commented):
Made the requested changes.

jtnydv commented:
> Thank you for your tutorial! You are actually installing the proprietary "OpenVPN Access Server" product and not the OSS OpenVPN server software. The tutorial should clearly reflect this.
> Can you please add small headlines about what you do in each step?
> Please also take a look at our [tutorial template](https://github.com/hetzneronline/community-content/blob/master/tutorial-template.md) about the formatting of each step. You should replace `### Step X` with `## Step X`.
> Your `NOTE` at the end should be a `Conclusion` instead.

Added the conclusion and changed the Markdown heading size.

rook59 reviewed (commented):
Thank you. Looks better to me now.
Last thing you would need to do is to remove the header_image tag and the header image itself.

rook59 commented:
@jtnydv can you please fix the last small things? afterwards it is ready to go

jtnydv commented:
Sorry for such delay. I've made the required changes as stated in the last comment. Please let me know if anything else is required of me. Thanks.

rook59 reviewed (approved):
Thanks

jtnydv commented:
Hey Team, is my article not eligible for the 50 USD Hetzner credits?

leojakeson commented:
Apologies for the late response, we just sent you an email with further instructions.
